lab/roles.py

```diff
@@ -119,11 +119,13 @@ def role(self, name, **kwargs):
     def _close_role(self, role):
         config = self.envmng.config
         config.hook.pytest_lab_role_destroyed(config=config, ctl=role)
-        try:
-            role.close()
-        except AttributeError:
-            pass
-        pytest.env.config.pluginmanager.unregister(plugin=role)
+        close = getattr(role, 'close', None)
+        if close:
+            close()
+            pytest.env.config.pluginmanager.unregister(plugin=role)
+        else:
+            self.log.warning(
+                "{} does not define a teardown `close()` method".format(role))
```

Comment:
Wait, why is this a warning now? I thought we had agreed on debug, and on putting it under both branches...